homeassistant/components/imap/__init__.py

```diff
@@ -188,6 +196,42 @@ async def async_delete(call: ServiceCall) -> None:
 
     hass.services.async_register(DOMAIN, "delete", async_delete, SERVICE_DELETE_SCHEMA)
 
+    async def async_fetch(call: ServiceCall) -> ServiceResponse:
+        """Process fetch email service and return content."""
+        entry_id: str = call.data[CONF_ENTRY]
+        uid: str = call.data[CONF_UID]
+        _LOGGER.debug(
+            "Fetch text for message %s. Entry: %s",
+            uid,
+            entry_id,
+        )
+        client = await async_get_imap_client(hass, entry_id)
```

Comment:
Maybe this should become an async context manager since its used in quite a few places?

Reply:
BTW it is intended to have a seperate session for the service calls apart from the base service of imap integration it self